### PR TITLE
Reduce use of memory in authorizer codepaths

### LINKED ISSUE #13316: potential performance cleanup of the authorizer
There were a couple comments about running toLower and copying a struct around https://github.com/openshift/origin/pull/13259#discussion_r105023565, https://github.com/openshift/origin/pull/13259#discussion_r105025947, and https://github.com/openshift/origin/pull/13259#discussion_r105028421.  

They could be cleaned up.

### PR DESCRIPTION
**Pull request by simo5**
Fixes: #13316

### CODE REVIEW

**Comment by simo5** [comment]
note that of the three problems reported in #13316 one is gone (the code is gone), one is addressed here (hopefully) and one is unfixable, as we use the kube authorizer interfaces which require we copy stuff into the struct @liggitt wasn't happy about.

@liggit @deads2k PTAL

**Comment by liggitt** [comment]
request info should not need normalization... I think we can directly compare

File: pkg/authorization/authorizer/personal_subjectaccessreview.go [hunk 1]
```diff
@@ -42,10 +42,10 @@ func (a *personalSARRequestInfoResolver) NewRequestInfo(req *http.Request) (*req
 	case len(requestInfo.Subresource) != 0:
 		return requestInfo, nil
 
-	case strings.ToLower(requestInfo.Verb) != "create":
+	case strings.EqualFold(requestInfo.Verb, "create") != true:
```
Comment (enj) [inline pkg/authorization/authorizer/personal_subjectaccessreview.go]: `!= true` ?

**Comment by enj** [comment]
> request info should not need normalization... I think we can directly compare

@deads2k I vaguely remember reviewing this code.  I believe we no longer need the normalization because RBAC is fully case sensitive?

**Comment by simo5** [comment]
A related PR is here https://github.com/kubernetes/kubernetes/pull/58356
There is no dependency on it, but things should improve once a rebase will pick that one up, assuming upstream will approve.

**Comment by simo5** [comment]
Flake #17556

**Comment by simo5** [comment]
@deads2k PTAL PR is ready to go

**Pull request merged**